### PR TITLE
Add ability to load dashboards for system.dashboards from config

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add ability to load dashboards for system.dashboards from config (once set, they overrides the default dashboards preset)

One of the obvious reasons is to allow rendering them with readonly user, which is not possible right now, due to usage of merge() function there.

Another one, is to add some custom metrics.

Follow-up for: https://github.com/ClickHouse/ClickHouse/pull/56771 (cc @serxa )

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 097aae0e0b2222fc92f939f43d193e6b6aee8e7c with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65555/097aae0e0b2222fc92f939f43d193e6b6aee8e7c/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Cloud fork sync (only for ClickHouse Inc. employees)</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by azat** [comment]
>Integration tests (aarch64) [5/6] — fail: 1, passed: 583


- `test_checking_s3_blobs_paranoid/test.py::test_when_s3_broken_pipe_at_upload_is_retried` is flaky

File: programs/server/config.xml [hunk 1]
```diff
@@ -1297,6 +1297,30 @@
         <ttl>event_date + INTERVAL 30 DAY</ttl>
     </blob_storage_log>
 
+    <!-- Configure system.dashboards for dashboard.html.
+
+         Should have the following parameters:
+         - seconds
+         - rounding
```
Comment (serxa) [inline programs/server/config.xml:1304]: It is good to mention they can have different custom parameters as well with these two

**Review by serxa — APPROVED** [review]

File: programs/server/config.xml (hunk 1 above)
Comment (azat) [reply programs/server/config.xml]: Updated

**Comment by azat** [comment]
Test failures does not looks related.

**Review by serxa — APPROVED** [review]

**Comment by azat** [comment]
@serxa so does this PR waits something from my side?

**Comment by serxa** [comment]
No, it just waits for CI to be green in private work as well

**Comment by serxa** [comment]
rebase is the thing "Cloud fork sync" cannot handle at the moment. And CI is now in a much better shape, could you please resubmit this PR? It is just too many red flags here, but the code is fine.

**Comment by azat** [comment]
>rebase is the thing "Cloud fork sync" cannot handle at the moment

I've heard that the conflict resolution in this case got lost, but I guess you are talking about different thing, does it simply do not detect rebases and do not run CI?

> And CI is now in a much better shape, could you please resubmit this PR? It is just too many red flags here, but the code is fine.

OK

**Pull request closed without merging**